### PR TITLE
feat(phirgen): tket to phir-specific gate conversion

### PR DESCRIPTION
**Pull request by qartik**

### CODE REVIEW

File: tests/sample_data.py
Comment (qartik) [inline tests/sample_data.py:1]: Sorted the test files by their file size to keep the largest test at the end.

File: pytket/phir/phirgen.py [hunk 1]
```diff
@@ -1,35 +1,210 @@
...
+
+            case RangePredicateOp():  # where the condition is a range
+                cond: dict[str, Any]
+                match cmd.op.lower, cmd.op.upper:
+                    case l, u if l == u:
+                        cond = {
+                            "cop": "==",
+                            "args": [cmd.args[0].reg_name, u],
+                        }
+                    case l, 4294967295:
```
Comment (peter-campora) [inline pytket/phir/phirgen.py:151]: Nit: can we alias the right number to something like `INTMAX`?

**Review by peter-campora — DISMISSED** [review]
LGTM other than that minor nit around the magic number for `INTMAX` for the range predicate case.

**Review by peter-campora — APPROVED** [review]
LGTM now!

**Review by Asa-Kosto-QTM — APPROVED** [review]

**Pull request merged**